src/js/models/layer.js

```diff
@@ -288,7 +288,8 @@ define(function (require, exports, module) {
                 isLinked: this.isLinked,
                 vectorMaskEnabled: this.vectorMaskEnabled,
                 vectorMaskEmpty: this.vectorMaskEmpty,
-                textWarningLevel: this.textWarningLevel
+                textWarningLevel: this.textWarningLevel,
+                smartObject: this.smartObject
```

Comment:
Yes. `LayerFace` component relies on the `Layer#face` attribute in the SCU (https://github.com/adobe-photoshop/spaces-design/blob/master/src/js/jsx/sections/layers/LayerFace.jsx#L62). I am adding the smart object so it will aware of the change in smart object type and update its icon accordingly.